diff --git a/codec/src/lib.rs b/codec/src/lib.rs
index 0c230e95500d43ad4ca9613112ebbb2093d9cc57..6348a6d744e8011981d110fc7543810f168b8e08 100644
--- a/codec/src/lib.rs
+++ b/codec/src/lib.rs
@@ -37,6 +37,9 @@ use thiserror::Error;
 use wezterm_term::color::ColorPalette;
 use wezterm_term::{Alert, ClipboardSelection, StableRowIndex, TerminalSize};
 
+mod vtabs;
+pub use vtabs::*;
+
 #[derive(Error, Debug)]
 #[error("Corrupt Response: {0}")]
 pub struct CorruptResponse(String);
@@ -503,6 +506,16 @@ pdu! {
     GetPaneDirection: 60,
     GetPaneDirectionResponse: 61,
     AdjustPaneSize: 62,
+    // Retired, never to be reused: 63, 64, 67, 68.
+    GetPaneCloseInfo: 65,
+    GetPaneCloseInfoResponse: 66,
+    ListAdoptablePanes: 69,
+    ListAdoptablePanesResponse: 70,
+    AdoptPane: 71,
+    AdoptPaneResponse: 72,
+    ReplacePane: 73,
+    GetPaneLocation: 74,
+    GetPaneLocationResponse: 75,
 }
 
 impl Pdu {
