diff --git a/config/src/ssh.rs b/config/src/ssh.rs
index f1cac1a9effb346c6f8d3fde49499285ce2634ac..0b11936e3bc0e842396c04dc1797da6a22dc4c9f 100644
--- a/config/src/ssh.rs
+++ b/config/src/ssh.rs
@@ -68,6 +68,10 @@ pub struct SshDomain {
     #[dynamic(default)]
     pub connect_automatically: bool,
 
+    /// Keep layouts on this mux, spawning remote shells without importing remote tabs.
+    #[dynamic(default)]
+    pub local_pane_layout: bool,
+
     #[dynamic(default = "default_read_timeout")]
     pub timeout: Duration,
 
diff --git a/config/src/tls.rs b/config/src/tls.rs
index 2c9adf57f8ec6a99b0938c4d02d8bb60e1851d7c..4d6bbd661f3a836b1aa81a0300e4c1abd0aab88f 100644
--- a/config/src/tls.rs
+++ b/config/src/tls.rs
@@ -76,6 +76,10 @@ pub struct TlsDomainClient {
     #[dynamic(default)]
     pub connect_automatically: bool,
 
+    /// Keep layouts on this mux, spawning remote shells without importing remote tabs.
+    #[dynamic(default)]
+    pub local_pane_layout: bool,
+
     #[dynamic(default = "default_read_timeout")]
     pub read_timeout: Duration,
 
diff --git a/config/src/unix.rs b/config/src/unix.rs
index 123691987ce30093bead1ccee0110c3145fcabd5..aa18e37caee16e2f0aa0b2c2570a3bca11d7ee94 100644
--- a/config/src/unix.rs
+++ b/config/src/unix.rs
@@ -20,6 +20,10 @@ pub struct UnixDomain {
     #[dynamic(default)]
     pub connect_automatically: bool,
 
+    /// Keep layouts on this mux, spawning remote shells without importing remote tabs.
+    #[dynamic(default)]
+    pub local_pane_layout: bool,
+
     /// If true, do not attempt to start this server if we try and fail to
     /// connect to it.
     #[dynamic(default)]
@@ -70,6 +74,7 @@ impl Default for UnixDomain {
             name: String::new(),
             socket_path: None,
             connect_automatically: false,
+            local_pane_layout: false,
             no_serve_automatically: false,
             serve_command: None,
             skip_permissions_check: false,
