diff --git a/wezterm-input-types/src/lib.rs b/wezterm-input-types/src/lib.rs
index f5673f600d8a06b4239a9a3c1146fc387123d8a8..8af3c434ac1c00009bd19b8506027924e2f19322 100644
--- a/wezterm-input-types/src/lib.rs
+++ b/wezterm-input-types/src/lib.rs
@@ -2122,7 +2122,11 @@ impl TryFrom<String> for WindowDecorations {
 
 impl Default for WindowDecorations {
     fn default() -> Self {
-        WindowDecorations::TITLE | WindowDecorations::RESIZE
+        if cfg!(target_os = "macos") {
+            Self::INTEGRATED_BUTTONS | Self::RESIZE
+        } else {
+            Self::TITLE | Self::RESIZE
+        }
     }
 }
 
